### DISCUSSION TITLE
How to use with PHP / Guzzle?

### DISCUSSION THREAD

**Discussion by paulvancity** [cloudbet/docs#69] [General]
Hello,

I am trying to do a simple PHP call with Guzzle and keep getting an error. 

Is there a PHP library offered that I can use? 

Here is my code below

`
//include latest guzzle lib
require_once '/var/www/begin.php';

define('CB_TOKEN', 'MY TOKEN');

use Guzzle\Http\Client;
use Guzzle\Http\Exception\RequestException;

// Define the base URI and headers
$baseUri = 'https://sports-api.cloudbet.com/pub/';
$headers = [
    'accept' => 'application/json',
    'X-API-Key' => CB_TOKEN,
];

// Create a new Guzzle client instance with default headers
$client = new Client(['base_uri' => $baseUri, 'headers' => $headers]);

try {
    // Send a GET request to the specified endpoint
    $response = $client->get('v2/odds/events', [
        'query' => [
            'sport' => 'basketball',
            'from' => 1693349945,
            'to' => 1693449945,
            'live' => 'false',
            'markets' => 'markets=basketball.totals',
            'limit' => 10000,
        ]
    ]);

    // Get the response body as a string
    $responseBody = $response->getBody()->getContents();

    // Handle the response body
    echo $responseBody;
} catch (RequestException $e) {
    // Handle exceptions and errors here
    echo 'Request Exception: ' . $e->getMessage();
}

/*
PHP Fatal error:  Uncaught TypeError: strpos(): Argument #1 ($haystack) must be of type string, array given in /var/www/sports.longtechnology.io/lib/vendor/guzzle/guzzle/src/Guzzle/Parser/UriTemplate/UriTemplate.php:41
Stack trace:
#0 /var/www/lib/vendor/guzzle/guzzle/src/Guzzle/Parser/UriTemplate/UriTemplate.php(41): strpos()
#1 /var/www/lib/vendor/guzzle/guzzle/src/Guzzle/Http/Client.php(356): Guzzle\Parser\UriTemplate\UriTemplate->expand()
#2 /var/www/lib/vendor/guzzle/guzzle/src/Guzzle/Http/Client.php(206): Guzzle\Http\Client->expandTemplate()
#3 /var/www/lib/vendor/guzzle/guzzle/src/Guzzle/Http/Client.php(186): Guzzle\Http\Client->getBaseUrl()
#4 /var/www/lib/vendor/guzzle/guzzle/src/Guzzle/Http/Client.php(242): Guzzle\Http\Client->createRequest()
#5 /var/www/admin/cloudbet.php(29): Guzzle\Http\Client->get()
#6 {main}
  thrown in /var/www/lib/vendor/guzzle/guzzle/src/Guzzle/Parser/UriTemplate/UriTemplate.php on line 41

*/
`

**Comment by kgravenreuth** [comment]
Hi @paulvancity 

While we don't have a PHP library or SDK for the Cloudbet API yet, we would encourage you to look at existing documentation and tutorials to get an idea of the responses that you should be expecting here. Furthermore, please note that the response is JSON encoded, which might need some handling on your part.

1. [How to build a betting bot with the Cloudbet API](https://bitedge.com/blog/how-to-build-a-betting-bot-with-the-cloudbet-api/) -> This is based on Python, which has similar dynamic language concepts as PHP
2. [Sports Betting API Tutorial With Golang](https://www.cloudbet.com/en/blog/posts/sports-betting-api-tutorial-with-golang)

Cheers,
Klaus Gravenreuth
Product & Engineering @ Cloudbet
